pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -477,3 +474,52 @@ func (m *manager) updateContainerCPUSet(containerID string, cpus cpuset.CPUSet)
 func (m *manager) GetCPUs(podUID, containerName string) cpuset.CPUSet {
 	return m.state.GetCPUSetOrDefault(podUID, containerName)
 }
+
+func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	klog.InfoS("CPUManager Admit Handler")
+	pod := attrs.Pod
+	return m.policy.Admit(pod)
+}
+
+func smtAlignmentError(requestedCPUs, cpusPerCore int) lifecycle.PodAdmitResult {
+	errorMessage := fmt.Sprintf("Number of CPUs requested should be a multiple of number of CPUs on a core = %d on this system. Requested CPU count = %d", cpusPerCore, requestedCPUs)
+	return lifecycle.PodAdmitResult{
+		Message: errorMessage,
+		Reason:  "SMTAlignmentError",
+		Admit:   false,
+	}
+}
+
+func admitPod() lifecycle.PodAdmitResult {
+	return lifecycle.PodAdmitResult{
+		Admit: true,
+	}
+}
```

Comment:
I don't think we want a new `Admit` handler here.

We can consider reworking the way `Allocate()` propagates errors to the existing `Admit` handler, but I'm not in favor of adding a new one like this.